### ISSUE TITLE
ImageChat results replication

### ISSUE THREAD

**Issue by DaraOrange** [facebookresearch/ParlAI#3579]
Hello! I am trying to train ImageChat model and get the same results as in the paper. I use the following command:
python3 ~/ParlAI/parlai/scripts/train_model.py \
--model projects.image_chat.transresnet_multimodal.transresnet_multimodal:TransresnetMultimodalAgent \
--task image_chat --model-file resnet152 --batchsize 128 --image_mode resnet152 \
--log_every_n_secs 120 --save_every_n_secs 600 --validation_every_n_epochs 2
but metrics are still too low (accuracy is around 0.23 instead of ~0.4).
@klshuster can you help please?
Also, how much epochs has training of your best model taken?

**Comment by klshuster** [comment]
hi there!

We found that the batchsize was quite important for maximizing performance; could you try setting the batchsize to 500, as we used in the paper? if not, try getting it as high as you can. Additionally, you might want to se the `--learningrate 0.0005`. 

As for epochs -> we would validate every 1 epoch, with `--validation-patience 10`

**Comment by DaraOrange** [comment]
Thanks, I will try it. Also you wrote in paper about pretrained transformer for context encoder initializing. Are its weights available?

**Comment by klshuster** [comment]
Yes, that is true (and might also be a major element of the performance drop). We do not provide the weights for the pre-trained transformer encoder from the paper; however, you could try using the reddit-pre-trained Bi-encoder from the [Poly-encoder](https://parl.ai/projects/polyencoder/) project. Specifically, you can do the following:

1. download the pretrained transformer via a command like this: `python3 /ParlAI/parlai/scripts/interactive.py -mf zoo:pretrained_transformers/bi_model_huge_reddit/model`
2. In a python shell, load the model from `ParlAI/data/models/pretrained_transformers/bi_model_huge_reddit/model`, extract the `context_encoder` (or `candidate_encoder`) from the `model['model']` state dict, and then save as a separate state dict
3. specify `--load-context-encoder-from /path/to/statedict` when training the transresnet model

Hope that helps!
